js/core/utils/deferred.d.ts

```diff
@@ -31,3 +31,14 @@ export interface PromiseType<T> { }
  */
 // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
 export type DxPromise<T = void> = {} extends PromiseType<T> ? Promise<T> : PromiseType<T>;
+
+/** @namespace DevExpress.core.utils */
+export interface PromiseExtension<T> {
```

Comment:
Since we no have an alias (`DxPromiseLike`) it's better make the extension a part of it:

```ts
export type DxPromiseLike<T> = DxPromise<T> & {
  then...
};
```

Reply:
done